File: lua/prime.lua
function sieve(size)
    if size <= 0 then return {} end
    local sieveArray = {false}
    for i = 2, size do sieveArray[i] = true end
    local root = math.sqrt(size) + 1
    for i = 2, root do
        if sieveArray[i] then
            for j = i * i, size, i do
                sieveArray[j] = false
            end
        end
    end
    return sieveArray
end

function countPrime(n)
    if n == 0 then return 0 end
    local sieveArray = sieve(n)
    local count = 0
    for i, p in ipairs(sieveArray) do
         if p then count = count + 1 end
    end
    return count
end

n = #arg >= 1 and tonumber(arg[1]) or 10000000
print(countPrime(n))

File: csharp/Prime.cs
using System;

class Prime {
    static bool[] Sieve(int size) {
    	bool[] sieveArray = new bool[size];
    	for (int i = 2; i < size; ++i) { sieveArray[i] = true; }
        int root = (int)Math.Sqrt(size) + 1;
        for (int i = 2; i < root; ++i) {
            if (!sieveArray[i]) continue;
            for (int j = i * i; j < size; j += i) {
                sieveArray[j] = false;
            }
        }
        return sieveArray;
    }

    static int CountPrime(int n) {
    	int count = 0;
    	foreach (bool x in Sieve(n + 1)) {
    		if (x) ++count;
    	}
        return count;
    }

    static void Main(string[] args) {
        int n = (args.Length >= 1) ? Int32.Parse(args[0]) : 10000000;
        Console.WriteLine(CountPrime(n));
    }
}
